mmv1/templates/terraform/pre_update/spanner_database.go.erb

```diff
@@ -1,29 +0,0 @@
-<%- # the license inside this block applies to this file
-	# Copyright 2022 Google Inc.
-	# Licensed under the Apache License, Version 2.0 (the "License");
-	# you may not use this file except in compliance with the License.
-	# You may obtain a copy of the License at
-	#
-	#     http://www.apache.org/licenses/LICENSE-2.0
-	#
-	# Unless required by applicable law or agreed to in writing, software
-	# distributed under the License is distributed on an "AS IS" BASIS,
-	# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-	# See the License for the specific language governing permissions and
-	# limitations under the License.
--%>
-
-if len(obj["statements"].([]string)) == 0 {
-	// Return early to avoid making an API call that errors,
-	// due to containing no DDL SQL statements
-	return resourceSpannerDatabaseRead(d, meta)
-}
```

Comment:
let's add back this file and add a conditional check `if obj["statements"] != nil`for the escape hatch. Plus, we also need to disable the partial mode to be able to write the vales to state by adding `d.Partial(false)` before `return resourceSpannerDatabaseRead(d, meta)`